db/compaction/compaction_iterator.cc

```diff
@@ -756,13 +756,19 @@ void CompactionIterator::NextFromInput() {
       }
 
       pinned_iters_mgr_.StartPinning();
+      Version* version = nullptr;
+      if (compaction_) {
+        version = compaction_->input_version();
+      }
+      BlobFetcher blob_fetcher(version, ReadOptions());
+
```

Comment:
I feel it would be nicer to just pass the `Version` to `MergeHelper::MergeUntil`. `MergeUntil` could then instantiate `BlobFetcher` internally if needed. (`MergeUntil` returns when it hits a blob, so any `BlobFetcher` object here is single-use.)